docs/decisions/0008-application_state_differenciation.rst

```diff
@@ -0,0 +1,177 @@
+
+See `0006-tanstack-react-query ADR <https://github.com/openedx/frontend-app-admin-portal/blob/master/docs/decisions/0006-tanstack-react-query.rst>`_ for more information
+
+As part of its implementation, a pattern has emerged of API data from the react-query cache is being populated within context providers. This has resulted in the unintended consequence of page refreshes when the top-level component context data is updated with the latest data from the ``react-query`` cache.
+
+The re-fetch behavior of ``react-query`` results in page components that aren't dependent on the updated context data to reload.
+
+The re-fetch behavior was occurring in the background of a component while the user is interacting with the page or if they had returned focus back onto the browser while interacting with other tabs or applications.
+
+An example where populating context data from ``react-query`` cache data resulted in unintentional behavior is the ``EnterpriseSubsidiesContext`` containing API data from the query key 'learner-credit-management.budgets', ``isFetchingBudgets`` attribute which has since been fixed.
+
+While the user was interacting with a ``FullscreenModal`` component and had not completed their intended activity within the modal before the global ``staleTime`` had expired, the re-fetch behavior would trigger resulting in a full page re-load, thus closing the modal without saving the users state information (form data etc.).
+
+Decisions
+*********
+
+We will take a more intentional approach when implementing API data fetching using the ``react-query`` in the following ways:
+
+#. Avoid adding API data fetched using ``react-query`` to a context layer
+    * If a context layer does require API data to exist in the context layer, do not use ``react-query`` to retrieve the data
+#. When using ``react-query`` make the call as close to the component that uses the returned response data
+    * This may require mapping a component from the parent component which completes the API call
+#. Leverage the `query key factory <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/learner-credit-management/data/constants.js#L67-L77>`_ setup to simplify boilerplate code when making the `useQuery` calls and `invalidateQueries`
+#. Implement a hook when calling the ``useQuery`` method to avoid duplicate code and easy maintainability. See this `hook <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/EnterpriseSubsidiesContext/data/hooks.js#L108-L120>`_ as boilerplate as an example, there are `others to reference in the repo <https://github.com/search?q=repo%3Aopenedx%2Ffrontend-app-admin-portal+%2FuseQuery%5C%28%2F&type=code>`_ as additional guides
+#. Invalidate queries when user interaction results in an API data change to avoid stale data
```

Comment:
Note, there is an alternative to query invalidation. If the change in server state is known, we can also consider manually update/modify the data stored in the query cache, where relevant, without needing to invalidate or re-fetch from the API, resulting in more performant UIs. Might be worth including that as an option here, too?